### DISCUSSION TITLE
Large discrepancy between DE Best-Fit and MCMC Parameter Space

### DISCUSSION THREAD

**Discussion by westobin** [phoebe-project/phoebe2#774] [Questions]
Hello!

I have a best-fit solution using the DE optimizer. It aligns to the data extremely well, yielding a reduced chi-squarred of 1.76. When I implement an MCMC process, one of the best-fit parameters (inclination) falls well outside the norm of the MCMC distribution. The MCMC means represent a much poorer fit with a reduced chi-squarred of 4.43.

DE solver ultimately resolved 10 parameters using a popsize=100. MCMC used 128 walkers and 10,000 iterations. From what I can tell, the resulting MCMC distributions are stable and only required a burn-in of about 600.

I have attached the corner plot with "Truths" set to the DE best-fit values.

I have also attached a light curve that shows the MCMC means model clearly deviates from both the data and the best-fit model at the light curve's eclipse minima.
 
![NSVS11868841Corner-HS-Truths](https://github.com/phoebe-project/phoebe2/assets/85952416/ff49039c-1789-49a8-9f2b-23d8d7a170b6)
![tlc-fvm](https://github.com/phoebe-project/phoebe2/assets/85952416/ef904ac9-09e7-4ddb-b082-2a2b625825d1)

I am not quite sure what I should do to resolve this discrepancy nor how to explain it. I don't understand why the parameter space for inclination appears to be offset, or why MCMC is pulling away from the best-fit rather than driving towards it.

I know that the MCMC parameter space may not perfectly match where an optimizer says is the best-fit, but it seems odd to me to be so far off for inclination while most of the other best-fit parameters are near the mode of their MCMC distributions. While this system doesn't have a total eclipse, we are using radial velocity data as well that shows very little discrepancy between the best-fit model and the MCMC means model.

Any thoughts?

- Wes

**Comment by westobin** [comment]
To check if the inclination parameter space is potentially just offset, I created two forward models from the MCMC means. The second model, I only changed the inclination to the DE-determined best-fit value. The result is a significantly Reduced-χ². In fact, the Reduced-χ² is actually BETTER than the DE-determined best-fit model:

- DE-determined best-fit model: Reduced-χ² = 1.77
- MCMC Means model: Reduced-χ² = 3.80
- MCMC Means model with corrected inclination: Reduced-χ² = 1.55

I am at a loss as to how to explain what exactly MCMC is doing... or why it's avoiding an inclination parameter that ostensibly improves the fit...

![tlc-i](https://github.com/phoebe-project/phoebe2/assets/85952416/6e35e0aa-3a5f-478b-bde7-34906d55d31c)

- Wes

**Comment by kecnry** [comment]
Hi Wes - where were your MCMC walkers initialized - did they include the "best solution" from DE?  Did you have any priors - and if so, were they the same for both the DE and MCMC runs?

The central values from all your MCMC posteriors also isn't necessarily going to give you the best merit function (as you noted), although I can't quite come up with a good explanation why inclination, in particular, would be so different while the others are not.  I wonder if it could be the "wrapping" nature (that inclination of 92 is really equivalent to 88 and that might be biasing the sampler in some way)?  PHOEBE does handle this to avoid getting a bimodal solution, but perhaps that is causing some issues.

But before we start chasing potential bugs - it might be interesting to pull the sample with the best lnprobability from the MCMC samples and see how that compares to both the posteriors and the DE solution.  And just to make sure we're comparing fairly, can you compare the lnprobabilities directly from phoebe (see [b.calculate_lnlikelihood](http://phoebe-project.org/docs/2.4/api/phoebe.parameters.ParameterSet.calculate_lnlikelihood) and add [b.calculate_lnp](http://phoebe-project.org/docs/2.4/api/phoebe.frontend.bundle.Bundle.calculate_lnp) if you have priors) rather than computing a reduced chi2?

**Comment by westobin** [comment]
Kyle,

Thanks for the response and suggestions. I think we have found the issue and it appears to be within our data processing apart from PHOEBE. As far as I can tell, PHOEBE and MCMC appear to be functioning properly based on the inputs we're providing.

- Wes
